lib/ruby_llm/provider.rb

```diff
@@ -78,13 +79,21 @@ def ensure_configured!
       end
 
       def sync_response(payload)
-        response = post completion_url, payload
+        model_id = payload[:model]
+        response = post completion_url, payload, model_id: model_id
         parse_completion_response response
       end
 
-      def post(url, payload)
+      def post(url, payload, model_id: nil)
+        request_headers = headers
+
+        if model_id && capabilities.respond_to?(:additional_headers_for_model)
+          additional_headers = capabilities.additional_headers_for_model(model_id)
+          request_headers = request_headers.merge(additional_headers) unless additional_headers.empty?
+        end
+
         connection.post url, payload do |req|
-          req.headers.merge! headers
+          req.headers.merge! request_headers
```

Comment:
changing the `post` method is a bit heavy handed. especially when the post needs to know about the `model_id` which is a clear violation of SRP